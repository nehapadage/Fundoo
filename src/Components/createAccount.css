
.maindiv1{
    background-color: white;
    /* text-align: center; */
    padding: 110px;
    width: 300px;
    height: 270px;
    margin: 430px;
    margin-top: 30px;
    border: 2px solid;
    border-color: grey;
    display: flex;
    flex-direction: column;
    /* flex-wrap: wrap; */
}

#fnln{
    margin-left: -85px; 
     display: flex;
    flex-direction: row; 
    justify-content: space-around; 
    /* height: auto;
    width: 300px; */
    
    /* background-color: thistle; */
}

#fnlnerr{
    margin-left: -85px; 
    display: flex;
   flex-direction: row; 
   justify-content: space-around;    
}

#passconerr{
    margin-left: -85px; 
    display: flex;
   flex-direction: row; 
   justify-content: space-around;    
}

#passcon{
    margin-left: -85px; 
    display: flex;
   flex-direction: row; 
   justify-content: space-between; 
}

#lastrow{
    margin-left: -85px; 
    display: flex;
    flex-direction: row;  
   justify-content: space-between; 
}

#login1{
    color: #1a73e8;
    background-color: transparent;
    display: flex;
    justify-content: flex-start;
    font-size: 15px;
}

/* #loginField{
height: 10px;
} */

/* #create{

} */

#email1{
    display: flex;  
    justify-content: flex-start;
    margin-left: -85px;
    /* width: 442px;
    height: 20px; */
    /* background-color: thistle; */
}

#emailText{
    width: 358px;
}



#fundoo1{
    font-size: xx-large;
    margin-top: -95px;
    /* margin-left: -280px; */
    display: flex;
    /* flex-direction: row; */
    margin-left: -85px;
    justify-content: flex-start;
}

#text1{
    color: #202124;
    font-family: 'Google Sans',arial,sans-serif;
    font-size: 24px;
        /* margin-left: -280px; */
        display: flex;
        /* flex-direction: row; */
        margin-left: -85px;
        justify-content: flex-start;

}

#text2{
    color:gray;
    font-family: 'Google Sans',arial,sans-serif;
    font-size: 15px;
        /* margin-left: -280px; */
        display: flex;
        /* flex-direction: row; */
        margin-left: -85px;
        justify-content: flex-start;

}

/* #firstName{
    display: flex;
    justify-content: flex-start;
} */

#flabel{
    color:blue;
}

#ulabel{
    color:red;
}

#nlabel{
    color:#FFC300 
}

#dlabel{
    color:blue;
}

#olabel{
    color:green;
}

#o2label{
    color:red;
}

#basic_adv_button{
     display:flex;
    flex-direction: row;
     width: 100%; 
     justify-content: space-between;
     margin-left: -28%;
}  

#bas{
    color: #1a73e8;
    background-color: transparent;
    display: flex;
    justify-content: flex-start;
    font-size: 15px; 
}

#adv{
    color: #1a73e8;
    background-color: transparent;
    display: flex;
    justify-content: flex-end;
    font-size: 15px; 
}

