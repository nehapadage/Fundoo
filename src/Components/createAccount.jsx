import React, { Component } from 'react'
import './createAccount.css'
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { createMuiTheme, MuiThemeProvider} from "@material-ui/core";
import { register } from '../services/userService'
import Snackbar from '@material-ui/core/Snackbar';

const theme = createMuiTheme({
    overrides: {
        MuiButton: {
            root: {
                textTransform: "lowercase",
                fontsize: 0.875
            }

        }
    }

});

class createAccount extends Component {
    constructor(props) {
        super(props);

        this.state = {
            firstName: "",
            lastName: "",
            email: "",
            password: "",
            confirmPassword: "",
            firstNameError: "",
            lastNameError: "",
            emailError: "",
            passwordError: "",
            confirmPasswordError: "",
            redirect: false,
            flag: false,
            flag1:false
        }
    }


    setRedirect = () => {
        this.setState({
            redirect: true
        })

    }

    renderRedirect = () => {
        if (this.state.redirect) {
            var path = '/login'
            this.props.history.push(path)
        }
    }

    handleLoginClick = () => {
        var path = '/login'
        this.props.history.push(path)
    }

    handlechangeall = (event) => {
        this.setState({ [event.target.name]: event.target.value })
    }

    validateEmail(email){
        // var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        var re=/^([a-z\d\.-]+)@([a-z\d-]+)\.([a-z]{2,8})(\.[a-z]{2,8})?$/
        return re.test(email);
      }
    

    validate = () => {
        let isError=false;
        const errors={

            firstNameError : "",
         lastNameError : "",
         emailError : "",
         passwordError : "",
         confirmPasswordError: "",

        };

    
    
        if (this.state.firstName.length < 4) {
            isError=true;
          errors.firstNameError = "First name needs to be atleast 5 characters long";
        }

        if (this.state.lastName.length < 4) {
            isError=true;
            errors.lastNameError = "Last name needs to be atleast 5 characters long";
          }
    
        // if (!this.state.email.includes("@")) {
        //     isError=true;
        //   errors.emailError = "Requires valid email";
        // }

        if (! this.validateEmail(this.state.email)) {
            isError = true;
            errors.emailError = "Requires valid email";
        }

        if ((this.state.password.length < 6 ) || (this.state.password.length > 12) ){
            isError=true;
            
            errors.passwordError = "Password length should greater than 6 and less than 12";
          }

          if ((this.state.confirmPassword.length < 6 ) || (this.state.confirmPassword.length > 12) ){
            isError=true;
            
            errors.confirmPasswordError = "Password length should greater than 6 and less than 12";
          }
    
          this.setState({
            ...this.state,
            ...errors
          });

          console.log("In validate----->"+this.state);
          
      
          return isError;
      };

      handleCreateAccountSubmit = (event) => {

        event.preventDefault();
       this.validate()
     const err = this.validate();

    //   if(this.state.password===this.state.confirmPassword)
    //   {
        var registerData = {};
        registerData.firstName = this.state.firstName;
        registerData.lastName = this.state.lastName;
        registerData.email = this.state.email;
        registerData.password = this.state.password
        registerData.service = 'basic'

        console.log("registerData--> ", JSON.stringify(registerData))



        register(registerData).then((res) => {
            console.log("respnse in register--> ", res)

            if (res.data.data.success === true) {
                this.setState({ flag1: true })
                // alert(`Registration Successful-----`);
                this.setRedirect();
            }
            // else {
            //     alert(`Email Already Exists-----`);
                
            // }
        }).catch((err) => {
            this.setState({ flag: true })
            console.log("error in registration--> ", err)
        })


    //   }
      

    

           
        

       if (!err) {
            // clear form
            this.setState({
              firstName: "", 
              firstNameError: "",
              lastName: "",
              lastNameError: "",
              email: "",
              emailError: "",
              password: "",
              passwordError: "",
              confirmPassword: "",
              confirmPasswordError: "",
            });
        }
        
        

    }

    handleClose = () => {
        console.log("Flag before",this.state.flag);
             
        this.setState({ flag: false})

        console.log("Flag before",this.state.flag);
    };

    handleClose1 = () => {
    
        console.log("Flag1 before",this.state.flag1);
        
        this.setState({flag1:false })


        console.log("Flag1 before",this.state.flag1);
    };

    render() {
        return (
            <MuiThemeProvider theme={theme}>
                <div className="maindiv1">
                {/* <form onSubmit={this.handlesubmit} ></form> */}
                <div id="fundoo1">
                    <label id="flabel">F</label>
                    <label id="ulabel">u</label>
                    <label id="nlabel">n</label>
                    <label id="dlabel">d</label>
                    <label id="olabel">o</label>
                    <label id="o2label">o</label>
                </div>
                <div id="text1">Create your Fundoo Account</div>

                <div id="fnln">
                    <div >
                    <TextField
                        id="firstName"
                        label="First name"
                        type="string"
                        name="firstName"
                        value={this.state.firstName}
                        // autoComplete="email"
                        margin="normal"
                        variant="outlined"
                        onChange={this.handlechangeall}
                        errorText={this.state.firstNameError}

                    />
                    </div>
                    &nbsp;&nbsp;&nbsp;
                    <div>
                    <TextField
                        id="lastName"
                        label="Last name"
                        type="string"
                        name="lastName"
                        value={this.state.lastName}
                        // autoComplete="email"
                        error={this.state.lastNameError}
                        margin="normal"
                        variant="outlined"
                        onChange={this.handlechangeall}
                        errorText={this.state.lastNameError}

                    />
                        </div> 
                                      
                   
                </div>

                <div id="fnlnerr">
                <div style={{ fontSize: 12, color: "red" }}>
            {this.state.firstNameError}
          </div>
          <div style={{ fontSize: 12, color: "red" }}>
            {this.state.lastNameError}
          </div>
                </div>
                <div id="email1">
                        <TextField
                            id="emailText"
                            label="Email Id"
                            type="email"
                            name="email"
                            value={this.state.email}
                            autoComplete="email"
                            margin="normal"
                            variant="outlined"
                            onChange={this.handlechangeall}
                            errorText={this.state.emailError}
                        />
                    </div>
                    <div style={{ fontSize: 12, color: "red" }}>
            {this.state.emailError}
          </div>
                    <div id="text2">You can use letters and numbers</div>

                    <div id="passcon">
                        <div>
                        <TextField
                            id="pass"
                            label="Password"
                            type="password"
                            name="password"
                            value={this.state.password}
                            autoComplete="current-password"
                            margin="normal"
                            variant="outlined"
                            onChange={this.handlechangeall}
                        />
                        </div>
                        &nbsp;&nbsp;&nbsp;
                        <div>
                        <TextField
                            id="conpass"
                            label="Confirm Password"
                            type="password"
                            name="confirmPassword"
                            value={this.state.confirmPassword}
                            autoComplete="current-password"
                            margin="normal"
                            variant="outlined"
                            onChange={this.handlechangeall}

                        />
                        </div>
                    </div>

                    <div id="passconerr">
                    <div style={{ fontSize: 12, color: "red" }}>
                        {this.state.passwordError}
                    </div>
                    <div style={{ fontSize: 12, color: "red" }}>
                        {this.state.confirmPasswordError}
                    </div>
                    </div>
                    <div id="lastrow">
                    
                        <Button id="login1"
                            onClick={this.handleLoginClick}
                        >Log in instead
                        </Button>
                    
                        <div>
                        {/* {this.renderRedirect()} */}
                            <Button
                                variant="contained"
                                color="primary"
                                onClick={this.handleCreateAccountSubmit}

                            >Create account
                                
                    </Button>
                        </div>
                    </div>

                     <Snackbar

                            anchorOrigin={{
                                vertical: 'bottom',
                                horizontal: 'center',
                            }}
                            open={this.state.flag}
                            autoHideDuration={6000}
                            onClose={this.handleClose}
                           
                            message="Registration Failed"
                       />

                       <Snackbar

                            anchorOrigin={{
                                vertical: 'bottom',
                                horizontal: 'center',
                            }}
                            open={this.state.flag1}
                            autoHideDuration={6000}
                            onClose={this.handleClose1}
                           
                            message="Registration Successful"
                       />
                        

            </div>
            </MuiThemeProvider>

                
            
            
        );
    }
};

export default createAccount;