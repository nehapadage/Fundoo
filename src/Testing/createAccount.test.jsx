import React from 'react';
import { shallow } from 'enzyme';
import createAccount from "../Components/createAccount"
// import renderer from 'react-test-renderer';
// import { mount, shallow, render } from â€˜enzyme';
// import createAccount from '../Components/createAccount'


describe('Register component tests', () => {

  const wrapper1 = shallow(<createAccount />);


  it('should render correctly in "debug" mode', () => {
    const component = shallow(<createAccount debug />);

    expect(component).toMatchSnapshot();
  });

//   it('should have an empty firstName, lastName,email and password state var', () => {
//     //Optionally test to check if password and email are empty strings on 
//     //  setup
//     expect(wrapper1.state('firstName')).toEqual('');
//     expect(wrapper1.state('lastName')).toEqual('');
//     expect(wrapper1.state('email')).toEqual('');
//     expect(wrapper1.state('password')).toEqual('');
    
//   });



//   it('register button status', () => {
//     // wrapper.find('Button').simulate('click');
//     expect(wrapper1.state('flag1')).toBe(false);
//   });












    it('should have input for email and password', ()=> {
      //Email and password input field should be present
      console.log("Length 1 is",wrapper1.find('EmailPass').length);
      
      expect(wrapper1.find('emailText')).toHaveLength(0);    // 1
      expect(wrapper1.find('fnfun')).toHaveLength(0);    // 1
      expect(wrapper1.find('lnfun')).toHaveLength(0);    // 2
      // expect('abc').toHaveLength(3);

  });



 
});

