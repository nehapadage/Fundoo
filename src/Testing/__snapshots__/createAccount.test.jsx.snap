// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component tests should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
